### ISSUE TITLE
test problem

### ISSUE THREAD

**Issue by Halfpoint1** [tsingqguo/exposure-fusion-shadow-removal#32]
I used the ISTD+ pre-trained model you provided to test, but I got different results from yours.

**Comment by fl82hope** [comment]
Could you provide more details?

**Comment by Halfpoint1** [comment]
Thank you for taking the time to answer my question. I used the shadow removal images you provided to calculate RMSE and got the same results as in your article. But I used the pth you provided, and run the test, and got a different RMSE. Looking forward to your answer. Thank you very much.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tsingqguo/exposure-fusion-shadow-removal"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月18日(星期四) 下午2:34
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32)





 
Could you provide more details?
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by fl82hope** [comment]
Could you provide your python and pytorch version of testing, also the test masks are generatd by our code?

**Comment by Halfpoint1** [comment]
My python version is 3.7.0 and pytorch version is 1.9.1.&nbsp;The test masks are in the attachment.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tsingqguo/exposure-fusion-shadow-removal"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月19日(星期五) 上午6:18
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32)





 
Could you provide your python and pytorch version of testing, also the test masks are generatd by our code?
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by fl82hope** [comment]
Sorry, I couldnot see the attachment, could you send me by email lanf@email.sc.edu.

**Comment by Halfpoint1** [comment]
please check

**Comment by yuanyufang** [comment]
> I used the ISTD+ pre-trained model you provided to test, but I got different results from yours.

请问你执行test的时候，有出现rmse为0的情况吗

**Comment by Halfpoint1** [comment]
他测rmse不是用的python代码，是用的matlab的一个代码，python代码没具体实现




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tsingqguo/exposure-fusion-shadow-removal"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年4月21日(星期四) 晚上8:27
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32)





  
I used the ISTD+ pre-trained model you provided to test, but I got different results from yours.
  
请问你执行test的时候，有出现rmse为0的情况吗

**Comment by yuanyufang** [comment]
> 他测rmse不是用的python代码，是用的matlab的一个代码，python代码没具体实现
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "tsingqguo/exposure-fusion-shadow-removal" ***@***.***&gt;; 发送时间:&nbsp;2022年4月21日(星期四) 晚上8:27 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32) I used the ISTD+ pre-trained model you provided to test, but I got different results from yours. 请问你执行test的时候，有出现rmse为0的情况吗 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
你好，我有点疑问想请教一下您。
1.运行sh OE_eval.sh文件，生成了如下结果，90-1.png和90-1-0.png这样的图，但是图中也有阴影，看着也不像是去阴影后的图片呀？
2.就是下图中rmse都是0,刚才你有说rmse是用matlab算的，我看到那个文件了，那
sh OE_eval.sh，是有什么用呢？出的是什么的效果图呢？
关于这两个问题我不是很懂，希望您能指导一下，不胜感激。
![68ebfb2ed2f8fefbad46e0f09997106](https://user-images.githubusercontent.com/94417754/164463246-779525da-537a-4416-a404-9925ed6806c0.jpg)

**Comment by Halfpoint1** [comment]
1.你可以对比一下原图，正常是去除阴影的图
2.oe_eval.sh 就是用来生成无阴影图像的


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tsingqguo/exposure-fusion-shadow-removal"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年4月21日(星期四) 晚上8:59
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32)





  
他测rmse不是用的python代码，是用的matlab的一个代码，python代码没具体实现
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "tsingqguo/exposure-fusion-shadow-removal" @.&gt;; 发送时间:&nbsp;2022年4月21日(星期四) 晚上8:27 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32) I used the ISTD+ pre-trained model you provided to test, but I got different results from yours. 请问你执行test的时候，有出现rmse为0的情况吗 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: @.***&gt;
 你好，我有点疑问想请教一下您。
 1.运行sh OE_eval.sh文件，生成了如下结果，90-1.png和90-1-0.png这样的图，但是图中也有阴影，看着也不像是去阴影后的图片呀？
 2.就是下图中rmse都是0,刚才你有说rmse是用matlab算的，我看到那个文件了，那
 sh OE_eval.sh，是有什么用呢？出的是什么的效果图呢？
 关于这两个问题我不是很懂，希望您能指导一下，不胜感激。

**Comment by yuanyufang** [comment]
好的 谢谢您

**Comment by yuanyufang** [comment]
> 他测rmse不是用的python代码，是用的matlab的一个代码，python代码没具体实现
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "tsingqguo/exposure-fusion-shadow-removal" ***@***.***&gt;; 发送时间:&nbsp;2022年4月21日(星期四) 晚上8:27 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32) I used the ISTD+ pre-trained model you provided to test, but I got different results from yours. 请问你执行test的时候，有出现rmse为0的情况吗 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
thank you answer very much much

**Comment by yuanyufang** [comment]
> 



> > 他测rmse不是用的python代码，是用的matlab的一个代码，python代码没具体实现
> > […](#)
> > ------------------ 原始邮件 ------------------ 发件人: "tsingqguo/exposure-fusion-shadow-removal" _**@**_._**>; 发送时间: 2022年4月21日(星期四) 晚上8:27 _**@**_.**_>; _**@**_._**_**@**_.**_>; 主题: Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32) I used the ISTD+ pre-trained model you provided to test, but I got different results from yours. 请问你执行test的时候，有出现rmse为0的情况吗 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: _**@**_.***>
> > thank you answer very much much

您好，打扰您一下，我有些问题想请教您一下。不胜感激！
1.我按照readme，先运行compute_params.py和test_mask_generation.py,得到曝光参数和mask，选用istd数据集，目录分为train-abc和test-abc,把model先改为fusion,运行sh OE_train.sh,跑400epoch,得到.pth文件和一些验证图（一排9张的那个）再改为refine，运行sh OE_train.sh，得到.pth文件和一些图（一排4张那种），然后修改OE_eval.sh里面的超参数，改为和train.sh的一样，再把model改为refine,载入的模型为训练refine得到的最后的.pth文件（训练refine的第400个epoch得到的.pth文件），然后运行sh OE_eval.sh得到一些测试结果，例如90-1和90-o.png文件，然后用作者提供的matlab，选用test_B（istd数据集中的数据）,test_C_fixed_official（istd+数据集中的数据）和刚生成的90-1.png，来算rmse.以上是我的步骤，
问题：
2.想请问您上面的步骤是否正确，就是我用的一些数据是否有问题？
3.上面步骤中，生成的mask为何没有用到呀？是不是在matlab中，test_B应该用生成的mask?
4.在matlab中，test_C_fixed_official是否是istd+数据集中的数据，因为我看名字是一样的，还是要用istd中的test_c?
5.运行sh OE_eval.sh得到一些测试结果，例如90-1和90-o.png文件,请问这两个图片有什么区别呢？
6.model改为fusion,运行sh OE_train.sh,跑400epoch,得到.pth文件，这些.pth文件后面是不是用不着了？因为运行sh OE_eval.sh，进行测试的时候，用的是Model为refine时跑出来的.pth文件。
7.我按照上面的步骤运行matlab跑出来的rmset挺大的，不知道博主你的结果是怎样的？下面两图分别是用测试结果图中，名字带o和不带o跑出来的结果。
![0adaf348dcaf77270dfcce43df1ff6c](https://user-images.githubusercontent.com/94417754/164967878-33210b68-cc64-4193-a5a3-c1a527242893.png)
![cc8f97cb9dad2fab42268134e2d5374](https://user-images.githubusercontent.com/94417754/164967901-2a817c26-1306-4f05-996a-3d134356a885.png)

**Comment by yuanyufang** [comment]
> Could you provide more details?

请问用matlab算rmse时，如果用istd数据集，无阴影图像是放test_c还是test_C_fixed_official，这一点我不是很理解

**Comment by yuanyufang** [comment]
> 他测rmse不是用的python代码，是用的matlab的一个代码，python代码没具体实现
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "tsingqguo/exposure-fusion-shadow-removal" ***@***.***&gt;; 发送时间:&nbsp;2022年4月21日(星期四) 晚上8:27 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32) I used the ISTD+ pre-trained model you provided to test, but I got different results from yours. 请问你执行test的时候，有出现rmse为0的情况吗 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

请问用matlab算rmse时，如果用istd数据集，无阴影图像是放test_c还是test_C_fixed_official，这一点我不是很理解

**Comment by Halfpoint1** [comment]
数据集分istd和istd+，作者用的istd+，你可以看下论文，是对istd颜色做了修正，也就是你说的test_C_fixed_official。测试的时候最好使用作者给的预训练模型。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tsingqguo/exposure-fusion-shadow-removal"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年4月25日(星期一) 中午11:25
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32)





  
他测rmse不是用的python代码，是用的matlab的一个代码，python代码没具体实现
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "tsingqguo/exposure-fusion-shadow-removal" @.&gt;; 发送时间:&nbsp;2022年4月21日(星期四) 晚上8:27 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32) I used the ISTD+ pre-trained model you provided to test, but I got different results from yours. 请问你执行test的时候，有出现rmse为0的情况吗 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: @.***&gt;
  
请问用matlab算rmse时，如果用istd数据集，无阴影图像是放test_c还是test_C_fixed_official，这一点我不是很理解

**Comment by yuanyufang** [comment]
> 数据集分istd和istd+，作者用的istd+，你可以看下论文，是对istd颜色做了修正，也就是你说的test_C_fixed_official。测试的时候最好使用作者给的预训练模型。
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "tsingqguo/exposure-fusion-shadow-removal" ***@***.***&gt;; 发送时间:&nbsp;2022年4月25日(星期一) 中午11:25 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32) 他测rmse不是用的python代码，是用的matlab的一个代码，python代码没具体实现 … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "tsingqguo/exposure-fusion-shadow-removal" @.&gt;; 发送时间:&nbsp;2022年4月21日(星期四) 晚上8:27 @.&gt;; @.@.&gt;; 主题:&nbsp;Re: [tsingqguo/exposure-fusion-shadow-removal] test problem (Issue #32) I used the ISTD+ pre-trained model you provided to test, but I got different results from yours. 请问你执行test的时候，有出现rmse为0的情况吗 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: @.***&gt; 请问用matlab算rmse时，如果用istd数据集，无阴影图像是放test_c还是test_C_fixed_official，这一点我不是很理解 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

ok 谢谢您我明白了，就是用istd的时候用test_c ,用istd+的时候用test_C_fixed_official，论文看过了，就是运行代码的时候有点不懂，还想问下博主，model改为fusion,运行sh OE_train.sh,跑400epoch,得到.pth文件，这些.pth文件后面是不是用不着了？因为运行sh OE_eval.sh，进行测试的时候，用的是Model为refine时跑出来的.pth文件，谢谢您！
